### ISSUE TITLE
config-integrity-guard: the only non-law key it tolerates is /^law-\d+$/, so a $comment in laws.notApplicable fails the audit and $-comments in ignores.byRule warn on every run

### ISSUE THREAD

**Issue by ruleofcodedev** [Rule-Of-Code/RoC#130]
**Version:** 7.21.0
**Law:** `config-integrity-guard` — `ConfigIntegrityGuardLaw` in `dist/checkers/meta-gate-laws/config-integrity-guard.js`, and `AuditEngine.validateByRuleKeys()` / `validateSeverityOverrides()` in `dist/core/auditing/audit-engine.js`

Kind: it fails a config that is correct — a false positive on a key that cannot be a law name in the first place.

## What happened

JSON has no comment syntax. The conventional way to annotate a JSON map is a `$`-prefixed key; `$comment` is a standard JSON Schema keyword. Adding one to `laws.notApplicable` — the section whose every entry is a human decision that needs explaining — fails the audit:

```
⚠️  laws.notApplicable: unknown law '$comment' — this waives nothing and the law still runs. Check the name with 'roc laws --list'.

❌ FAILED CONSTITUTIONAL LAWS:

1. Config Integrity Guard [1 violations] (Score: 90/100)
   • laws.notApplicable: unknown law '$comment' — this entry gates nothing; check the name with 'roc laws --list'

✗ Failed: 1
```

The same keys in `ignores.byRule` only warn, but they warn on every single run:

```
⚠️  ignores.byRule: unknown law '$comment' — these patterns filter nothing. Check the name with 'roc laws --list'.
⚠️  ignores.byRule: unknown law '$comment-logger' — these patterns filter nothing. Check the name with 'roc laws --list'.
⚠️  ignores.byRule: unknown law '$comment-origins' — these patterns filter nothing. Check the name with 'roc laws --list'.
⚠️  ignores.byRule: unknown law '$comment-coverage' — these patterns filter nothing. Check the name with 'roc laws --list'.
⚠️  ignores.byRule: unknown law '$comment-error-handling' — these patterns filter nothing. Check the name with 'roc laws --list'.
```

## What is wrong

Both code paths accept exactly one non-law key shape, `/^law-\d+$/`, and treat everything else as a law name that failed to resolve.

`dist/checkers/meta-gate-laws/config-integrity-guard.js:38` — this one is a **violation**, and it covers three sections (`ConfigIntegrityGuardLaw.SECTIONS = ['severity', 'notApplicable', 'enabled']`):

```js
for (const key of Object.keys(map)) {
    if (/^law-\d+$/.test(key)) continue;          // the only hatch
    const resolves = counts.get(key) ?? 0;
    if (resolves === 0) {
        violations.push(`laws.${section}: unknown law '${key}' — this entry gates nothing; ...`);
    }
```

`dist/core/auditing/audit-engine.js:302` — same shape, one severity lower, for `ignores.byRule` and `includes.byRule`:

```js
for (const key of Object.keys(section)) {
    if (/^law-\d+$/.test(key)) continue;
    if (known.has(key)) continue;
    console.warn(`⚠️  ${label}: unknown law '${key}' — these patterns filter nothing. ...`);
}
```

Neither file contains any `$` handling:

```
$ grep -c "startsWith('\$'" dist/checkers/meta-gate-laws/config-integrity-guard.js dist/core/auditing/audit-engine.js
dist/checkers/meta-gate-laws/config-integrity-guard.js:0
dist/core/auditing/audit-engine.js:0
```

## Reproduction

A/B against the same repo and the same command, with a control arm — the two config files differ by exactly one key, confirmed with a diff rather than by eye:

```
$ diff control.config.json probe.config.json
@@ -167,3 +167,4 @@
-      "core-web-vitals-compliance": "PARKED, ..."
+      "core-web-vitals-compliance": "PARKED, ...",
+      "$comment": "Why the entries below are parked — a JSON comment, not a law."
```

```
$ npx ruleofcode audit --mode=full --config probe.config.json
1. Config Integrity Guard [1 violations] (Score: 90/100)
✗ Failed: 1
$ echo $?
1

$ npx ruleofcode audit --mode=full --config control.config.json
103. Config Integrity Guard (Score: 100/100)
✗ Failed: 0
$ echo $?
0
```

One JSON comment is the entire difference between a green audit and a red one.

## What it cost me

The warning block is genuinely valuable — in the very same output it correctly caught four *stale* names in this config (`code-complexity`, `dead-code`, `angular-onpush-strategy`, `ngrx_d494c2aa`, all of which really do filter nothing). Five permanent false positives sitting in the same block is what taught me to skim it, and skimming it is how those four survived. That is the actual cost: the false ones are not merely noise, they are camouflage for the true ones.

The need for a comment is real enough that this config already works around it: a key named `comment_baseline_origin_2026-08-18` lives inside `thresholds.git` — which this validation does not reach — and holds three paragraphs on why the commit baseline is `origin/develop`. The same reasoning cannot be written next to the parked laws it is about.

Renaming a `$comment` to `law-0` does silence the check, but a reader then has to work out that it is not a law entry, which is worse than the warning.

## Is the same logic somewhere else?

Checked. `/^law-\d+$/` appears in exactly these two files, and they are the two paths above. `AuditEngine.warnOnSeverityValue()` is a separate check on the *value* and is correct — a `$comment` key never reaches it with a bad severity. `thresholds.*` is not covered by either path, which is why the workaround above is possible at all.

## Suggested fix

1. Skip `$`-prefixed keys alongside the existing hatch, in both files:

   ```js
   if (key.startsWith('$')) continue;
   ```

2. `$` cannot collide with a real law key — measured against the registry rather than assumed:

   ```js
   const { ModularLawsRegistry: R } = require('ruleofcode/dist/registry/modular-laws-registry.js');
   const keys = new Set();
   for (const l of R.getAll()) for (const k of R.lawIdentityKeys(l)) keys.add(k);
   // laws: 169 | identity keys: 695 | starting with '$': 0 | containing '$': 0
   ```

3. If comments are meant to be unsupported instead, the `laws.*` case still wants softening — `notApplicable` is the one section where the *reason* is the whole point of the entry, and today there is nowhere to put it.

## What I did NOT verify

- Whether this reproduces on 7.22.0. This project is pinned to 7.21.0 and I have not re-staged on the latest.
- Whether `includes.byRule` behaves identically. It shares the same loop and the same `sections` array, so I expect it to, but this config has no `includes.byRule` and I did not construct one to confirm it.
- Whether `laws.enabled` fails the same way. It is in `SECTIONS` alongside `notApplicable`, so it should, but I only exercised `notApplicable`.
- Whether any *other* config key namespace (`thresholds.*`, `ignores.paths`) validates keys somewhere else in the codebase. I grepped only the two paths that produce this message.

## My config

```json
"laws": {
  "notApplicable": { "...": "18 entries" },
  "severity": { "...": "169 entries" }
},
"ignores": {
  "byRule": {
    "$comment": "...",
    "$comment-logger": "...",
    "$comment-origins": "...",
    "$comment-coverage": "...",
    "$comment-error-handling": "..."
  }
}
```

**Comment by ruleofcodedev** [comment]
Fixed in #135, merged to `develop`. Ships in the next release.

`isNonLawConfigKey()` now lives in the module that already owns law-name identity: a key that cannot be a law name is not a typo'd law name.

Swept, not patched at the reported site. The same literal regex was tested in **four** places:

- `config-integrity-guard.ts:68` — the one you hit
- `audit-engine.ts:442` — severity overrides
- `audit-engine.ts:491` — unknown-law warnings
- `audit-engine.ts:541` — `notApplicable` waivers

All four call the predicate now, so you will not find the second site next week.

The red control in the regression suite is the one that matters: a genuinely misspelled law name is still reported. Silencing typos in order to accept annotations would have traded this defect for a worse one.
